File: study.js
// Sample flashcards (question:answer)
let flashcards = [
  { question: "What is the capital of France?", answer: "Paris" },
  { question: "What is 2 + 2?", answer: "4" },
  { question: "What color is the sky?", answer: "Blue" }
];

let remainingCards = [];
let currentCard = null;
let correct = 0;
let incorrect = 0;

document.addEventListener("DOMContentLoaded", () => {
  document.getElementById("loadBtn").addEventListener("click", loadFlashcards);
  document.getElementById("nextBtn").addEventListener("click", nextCard);
  document.getElementById("checkBtn").addEventListener("click", checkAnswer);
});

function loadFlashcards() {
  remainingCards = [...flashcards];
  shuffleArray(remainingCards);
  correct = 0;
  incorrect = 0;
  nextCard();
}

function nextCard() {
  const questionLabel = document.getElementById("questionLabel");
  const answerInput = document.getElementById("answerInput");
  const output = document.getElementById("outputArea");

  if (remainingCards.length === 0) {
    questionLabel.textContent = "";
    output.value = `Quiz complete!\nCorrect: ${correct} | Incorrect: ${incorrect}`;
    return;
  }

  currentCard = remainingCards.pop();
  questionLabel.textContent = currentCard.question;
  answerInput.value = "";
  output.value = "";
}

function checkAnswer() {
  const answerInput = document.getElementById("answerInput");
  const output = document.getElementById("outputArea");

  if (!currentCard) {
    output.value = "No card selected.";
    return;
  }

  const userAnswer = answerInput.value.trim().toLowerCase();
  const correctAnswer = currentCard.answer.trim().toLowerCase();

  if (userAnswer === correctAnswer) {
    correct++;
    output.value = "Correct!";
  } else {
    incorrect++;
    output.value = `Incorrect. The correct answer was: ${currentCard.answer}`;
  }

  output.value += `\nCorrect: ${correct} | Incorrect: ${incorrect}`;
}

function shuffleArray(array) {
  for (let i = array.length - 1; i > 0; i--) {
    const j = Math.floor(Math.random() * (i + 1));
    [array[i], array[j]] = [array[j], array[i]];
  }
}

document.addEventListener("DOMContentLoaded", () => {
  // Button and body references
  const toggleBtn = document.getElementById("toggleMode");
  const body = document.body;

  // Load and apply saved theme
  const savedTheme = localStorage.getItem("theme") || "light";
  body.classList.add(savedTheme);
  updateToggleText(savedTheme);

  toggleBtn.addEventListener("click", () => {
    const current = body.classList.contains("dark") ? "dark" : "light";
    const newTheme = current === "dark" ? "light" : "dark";
    body.classList.replace(current, newTheme);
    localStorage.setItem("theme", newTheme);
    updateToggleText(newTheme);
  });

  function updateToggleText(theme) {
    toggleBtn.textContent = theme === "dark" ? "🌞 Toggle Light Mode" : "🌙 Toggle Dark Mode";
  }
});

File: study.py
from BreezyPythonGUI import EasyFrame
from tkinter.filedialog import askopenfilename
import random

class FlashcardApp(EasyFrame):
    def __init__(self):
        EasyFrame.__init__(self, title="Flashcard Quiz App")

        self.addLabel(text="Question:", row=0, column=0)
        self.questionLabel = self.addLabel(text="", row=0, column=1)

        self.addLabel(text="Your Answer:", row=1, column=0)
        self.answerField = self.addTextField("", row=1, column=1, width=40)

        self.addButton("Check Answer", 2, 0, self.checkAnswer)
        self.addButton("Next Card", 2, 1, self.nextCard)
        self.addButton("Load Flashcards", 3, 0, self.loadFlashcards)

        self.outputArea = self.addTextArea("", row=5, column=0, width=50, height=4)
        self.outputArea["state"] = "disabled"

        self.flashcards = []
        self.remainingCards = []
        self.currentCard = None
        self.correct = 0
        self.incorrect = 0

    def loadFlashcards(self):
        filename = askopenfilename(title="Select flashcard file")
        if filename:
            with open(filename, "r") as file:
                self.flashcards = []
                for line in file:
                    if ':' in line:
                        q, a = line.strip().split(":", 1)
                        self.flashcards.append((q.strip(), a.strip()))
            random.shuffle(self.flashcards)
            self.remainingCards = self.flashcards[:]
            self.correct = 0
            self.incorrect = 0
            self.nextCard()

    def nextCard(self):
        if not self.remainingCards:
            self.showOutput("No more flashcards. Quiz complete!\nCorrect: {} | Incorrect: {}".format(self.correct, self.incorrect))
            self.questionLabel["text"] = ""
            return
        self.currentCard = self.remainingCards.pop()
        self.questionLabel["text"] = self.currentCard[0]
        self.answerField.setValue("")
        self.showOutput("")

    def checkAnswer(self):
        if not self.currentCard:
            self.showOutput("No card selected.")
            return
        userAnswer = self.answerField.get().strip()
        correctAnswer = self.currentCard[1]
        if userAnswer.lower() == correctAnswer.lower():
            self.correct += 1
            result = "Correct!"
        else:
            self.incorrect += 1
            result = f"Incorrect. The correct answer was: {correctAnswer}"
        score = f"Correct: {self.correct} | Incorrect: {self.incorrect}"
        self.showOutput(f"{result}\n{score}")

    def showOutput(self, message):
        self.outputArea["state"] = "normal"
        self.outputArea.delete("1.0", "end"); self.outputArea.insert("end", message)
        self.outputArea["state"] = "disabled"

if __name__ == "__main__":
    FlashcardApp().mainloop()
